### PR TITLE
Add the ability to post to introspection endpoint.

### PR DESCRIPTION
**Pull request by howbazaar**
This adds post to the juju-introspect CLI command. 

The worker itself was cleaned up but doesn't yet use the post functionality.
That is coming in a follow up branch.

## QA steps

Just the unit tests at this stage.

### CODE REVIEW

File: cmd/jujud/introspect/introspect_test.go [hunk 1]
```diff
@@ -125,6 +127,42 @@ argh
...
+	listener, err := net.Listen("unix", "@"+filepath.Join(config.DataDir, "jujud-machine-0"))
+	c.Assert(err, jc.ErrorIsNil)
+	defer listener.Close()
+
+	srv := newServer(listener)
+	go srv.Serve(listener)
+	defer srv.Shutdown(context.Background())
+
+	ctx, err := s.run(c, "postonly", "--agent=machine-0")
+	c.Assert(err, gc.ErrorMatches, `response returned 405 \(Method Not Allowed\)`)
```
Comment (tlm) [inline cmd/jujud/introspect/introspect_test.go:140]: Knit pick " over `

Comment (howbazaar) [reply cmd/jujud/introspect/introspect_test.go:140]: Fixed.

**Comment by howbazaar** [comment]
$$merge$$

**Comment by tlm** [comment]
LGTM

**Comment by howbazaar** [comment]
$$merge$$

**Comment by hpidcock** [comment]
```
13:51:31 FAIL: github.com/juju/juju/apiserver/introspection_test.go:35: introspectionSuite.TestAccess
13:51:31 
13:51:31 [LOG] 0:00.002 INFO juju.state using client-side transactions
13:51:31 [LOG] 0:00.092 INFO juju.state controller settings not found, early stage initialization assumed
13:51:31 [LOG] 0:00.092 DEBUG juju.state mongodb initialised
13:51:31 [LOG] 0:00.092 INFO juju.state starting standard state workers
13:51:31 [LOG] 0:00.092 INFO juju.state creating cloud image metadata storage
13:51:31 [LOG] 0:00.092 INFO juju.state started state for model-f4d28f8a-2235-45d2-811f-c9e6d8341a33 successfully
13:51:31 [LOG] 0:00.093 INFO juju.state initializing controller model f4d28f8a-2235-45d2-811f-c9e6d8341a33
13:51:31 [LOG] 0:00.149 INFO juju.state creating logs collection for f4d28f8a-2235-45d2-811f-c9e6d8341a33, capped at 1 MiB
13:51:31 [LOG] 0:00.175 DEBUG juju.state.bakerystorage using local users third party key: wb9rjdEvuSsUsb8sepoRKrKR63wvQ2Jz29EiblHH4n0=
13:51:31 [LOG] 0:00.176 DEBUG juju.state.bakerystorage using local users key: aPoM1ofVU7AFptF2tKPt3pDXKePORDRUDJ0OL/XVPFs=
13:51:31 [LOG] 0:00.186 INFO juju.state.allwatcher allwatcher loaded for model "f4d28f8a-2235-45d2-811f-c9e6d8341a33" in 0s
13:51:31 started HTTP server listening on "127.0.0.1:45087"
13:51:31 [LOG] 0:00.189 DEBUG juju.state.bakerystorage using offers third party key: zKxzanqt7WC8loRjUtqiVtFZ+8PtFZAx+TFb9psv5XY=
13:51:31 [LOG] 0:00.189 WARNING juju.apiserver.logsink Unable to prime /tmp/check-6415295772277131806/2/logsink.log (proceeding anyway): chown /tmp/check-6415295772277131806/2/logsink.log: operation not permitted
13:51:31 sendRequest: https://127.0.0.1:45087/introspection/navel
13:51:31 github.com/juju/juju/apiserver/introspection_test.go:36:
13:51:31     ...open github.com/juju/juju/apiserver/introspection_test.go: no such file or directory
13:51:31 github.com/juju/juju/apiserver/introspection_test.go:59:
13:51:31     ...open github.com/juju/juju/apiserver/introspection_test.go: no such file or directory
13:51:31 ... obtained int = 404
13:51:31 ... expected int = 200
13:51:31 
13:51:31 [LOG] 0:00.228 DEBUG juju.state closed state without error
13:51:31 
13:51:31 ----------------------------------------------------------------------
13:51:31 FAIL: github.com/juju/juju/apiserver/introspection_test.go:65: introspectionSuite.TestAccessDenied
13:51:31 
13:51:31 [LOG] 0:00.002 INFO juju.state using client-side transactions
13:51:31 [LOG] 0:00.042 INFO juju.state controller settings not found, early stage initialization assumed
13:51:31 [LOG] 0:00.042 DEBUG juju.state mongodb initialised
13:51:31 [LOG] 0:00.042 INFO juju.state starting standard state workers
13:51:31 [LOG] 0:00.042 INFO juju.state creating cloud image metadata storage
13:51:31 [LOG] 0:00.042 INFO juju.state started state for model-1732d593-910b-47fa-86f4-aaa36432cd02 successfully
13:51:31 [LOG] 0:00.049 INFO juju.state initializing controller model 1732d593-910b-47fa-86f4-aaa36432cd02
13:51:31 [LOG] 0:00.106 INFO juju.state creating logs collection for 1732d593-910b-47fa-86f4-aaa36432cd02, capped at 1 MiB
13:51:31 [LOG] 0:00.109 DEBUG juju.state.bakerystorage using local users third party key: +8IjwNIGKI3QNk6EHfOd58q5drFG3pM4jcJsKALGw20=
13:51:31 [LOG] 0:00.109 DEBUG juju.state.bakerystorage using local users key: oP/TDZmAuWw4/YcJA3zFpzFr1dqk4uQ0cZzoN6VL5DI=
13:51:31 [LOG] 0:00.116 INFO juju.state.allwatcher allwatcher loaded for model "1732d593-910b-47fa-86f4-aaa36432cd02" in 0s
13:51:31 started HTTP server listening on "127.0.0.1:45685"
13:51:31 [LOG] 0:00.119 DEBUG juju.state.bakerystorage using offers third party key: OKr1aC/nO8/lVRqtiwgE7Ot8B/xtPvSTPu+pb7sICzk=
13:51:31 [LOG] 0:00.120 WARNING juju.apiserver.logsink Unable to prime /tmp/check-6415295772277131806/5/logsink.log (proceeding anyway): chown /tmp/check-6415295772277131806/5/logsink.log: operation not permitted
13:51:31 sendRequest: https://127.0.0.1:45685/introspection/navel
13:51:31 github.com/juju/juju/apiserver/introspection_test.go:73:
13:51:31     ...open github.com/juju/juju/apiserver/introspection_test.go: no such file or directory
13:51:31 ... obtained int = 404
13:51:31 ... expected int = 403
13:51:31 
13:51:31 [LOG] 0:00.170 DEBUG juju.state closed state without error
```

**Comment by howbazaar** [comment]
Yeah, I've realised that I can't just remove the apiserver access. At the very least, we need to export the prometheus metrics there, but it makes sense to limit it to that, and not offer all the same access points to both.

**Comment by howbazaar** [comment]
$$merge$$

**Pull request merged**